### PR TITLE
Add support for CPU sockets and threads

### PR DESCRIPTION
**Pull request by ksimon1**
Add support for sockets and threads in templates

### CODE REVIEW

**Comment by ovirt-infra** [comment]
Hello contributor, thanks for submitting a PR for this project! 

I am the bot who triggers "standard-CI" builds for this project.
As a security measure, I will not run automated tests on PRs that are not from white-listed contributors.

In order to allow automated tests to run, please ask one of the project maintainers to review the code and then do one of the following:

1. Type `ci test please` on this PR to trigger automated tests for it.
2. Type `ci add to whitelist` on this PR to trigger automated tests for it and also add you to the contributor white-list so that your future PRs will be tested automatically.
3. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the project organization, so you'll be able to run tests for all the organization's projects.

**Comment by MarSik** [comment]
ci add to whitelist

File: generate-templates.yaml [hunk 1]
```diff
@@ -12,14 +12,14 @@
...
-    - {flavor: large, workload: generic, memsize: "8G", cores: 2, iothreads: False}
-    - {flavor: large, workload: highperformance, memsize: "8G", cores: 2, iothreads: True}
+    - {flavor: tiny, workload: generic, memsize: "1G", cores: 1, sockets: 1, threads: 1, iothreads: False}
+    - {flavor: tiny, workload: highperformance, memsize: "1G", cores: 1, sockets: 1, threads: 1, iothreads: True}
+    - {flavor: small, workload: generic, memsize: "2G", cores: 1, sockets: 1, threads: 1, iothreads: False}
+    - {flavor: small, workload: highperformance, memsize: "2G", cores: 1, sockets: 1, threads: 1, iothreads: True}
+    - {flavor: medium, workload: generic, memsize: "4G", cores: 1, sockets: 1, threads: 1, iothreads: False}
+    - {flavor: medium, workload: highperformance, memsize: "4G", cores: 1, sockets: 1, threads: 1, iothreads: True}
+    - {flavor: large, workload: generic, memsize: "8G", cores: 2, sockets: 1, threads: 1, iothreads: False}
+    - {flavor: large, workload: highperformance, memsize: "8G", cores: 2, sockets: 1, threads: 1, iothreads: True}
```
Comment (MarSik) [inline generate-templates.yaml]: Please change the configuration of all OSes so it uses sockets instead of cores.

File: templates/_linux.yaml [hunk 2]
```diff
@@ -44,6 +44,10 @@ objects:
 
           cpu:
             cores: {{ item.cores }}
+{% if not osinfoname | regex_search('win') %}
```
Comment (MarSik) [inline templates/_linux.yaml]: What is the meaning of this? The include here is only used for linux guests.

Comment (MarSik) [reply templates/_linux.yaml]: Windows has its own template file (so does ubuntu btw, make sure it uses sockets and threads as well).

Comment (ksimon1) [reply templates/_linux.yaml]: According to this line https://github.com/kubevirt/common-templates/blob/master/templates/ubuntu.tpl.yaml#L17 Ubuntu should use generic template as well. Or am i wrong?

Comment (MarSik) [reply templates/_linux.yaml]: No you are right, Ubuntu is fine then.

**Comment by MarSik** [comment]
ci test please

**Comment by MarSik** [comment]
There is an issue with the test script, kubevirt changed namespace from kube-system to kubevirt in the new version. We have to adapt to it. @ksimon1 can you please add one more patch that fixes the test scripts and Makefile?

File: templates/_linux.yaml [hunk 3]
```diff
@@ -43,7 +45,8 @@ objects:
 {% endif %}
 
           cpu:
-            cores: {{ item.cores }}
+            sockets: {{ item.sockets }}
```
Comment (MarSik) [inline templates/_linux.yaml]: Here I would define all three topology attributes this way (for now):

```suggestion
            sockets: {{ item.cpus }}
            cores: 1
            threads: 1
```

Also do not forget to change this in the windows template as well.

File: generate-templates.yaml [hunk 4]
```diff
@@ -13,14 +13,14 @@
...
     with_items:
-    - {flavor: tiny, workload: generic, memsize: "1G", cores: 1, iothreads: False}
-    - {flavor: tiny, workload: highperformance, memsize: "1G", cores: 1, iothreads: True}
-    - {flavor: small, workload: generic, memsize: "2G", cores: 1, iothreads: False}
-    - {flavor: small, workload: highperformance, memsize: "2G", cores: 1, iothreads: True}
-    - {flavor: medium, workload: generic, memsize: "4G", cores: 1, iothreads: False}
-    - {flavor: medium, workload: highperformance, memsize: "4G", cores: 1, iothreads: True}
-    - {flavor: large, workload: generic, memsize: "8G", cores: 2, iothreads: False}
-    - {flavor: large, workload: highperformance, memsize: "8G", cores: 2, iothreads: True}
+    - {flavor: tiny, workload: generic, memsize: "1G", sockets: 1, threads: 1, iothreads: False}
```
[the remark this answers is not in the document]
Comment (ksimon1) [reply generate-templates.yaml]: because when we discussed it month ago, you said to remove cores.

**Pull request merged**